examples/features/advancedtls/server/main.go

```diff
@@ -0,0 +1,188 @@
+	identityProvider := makeIdentityProvider(useRevokedCert, credsDirectory)
+	defer identityProvider.Close()
+
+	rootProvider := makeRootProvider(credsDirectory)
+	defer rootProvider.Close()
+
+	crlProvider := makeCrlProvider(filepath.Join(credsDirectory, "crl"))
+	defer crlProvider.Close()
+
+	options := &advancedtls.Options{
+		IdentityOptions: advancedtls.IdentityCertificateOptions{
+			IdentityProvider: identityProvider,
+		},
+		RootOptions: advancedtls.RootCertificateOptions{
+			RootProvider: rootProvider,
+		},
+		RequireClientCert: true,
+		VerificationType:  advancedtls.CertVerification,
+	}
+
+	options.RevocationOptions = &advancedtls.RevocationOptions{
+		CRLProvider: crlProvider,
+	}
+
+	serverTLSCreds, err := advancedtls.NewServerCreds(options)
+	if err != nil {
+		fmt.Printf("Error %v\n", err)
+		os.Exit(1)
+	}
+
+	s := grpc.NewServer(grpc.Creds(serverTLSCreds))
+	lis, err := net.Listen("tcp", fmt.Sprintf(":%d", port))
+	if err != nil {
+		fmt.Printf("Failed to listen: %v\n", err)
+	}
+	name := "Good TLS Server"
+	if useRevokedCert {
+		name = "Revoked TLS Server"
+	}
+	pb.RegisterEchoServer(s, &server{name: name})
+	if err := s.Serve(lis); err != nil {
+		fmt.Printf("Failed to serve: %v\n", err)
+		os.Exit(1)
+	}
+
+}
+
+func makeRootProvider(credsDirectory string) certprovider.Provider {
+	rootOptions := pemfile.Options{
+		RootFile:        filepath.Join(credsDirectory, "/ca_cert.pem"),
+		RefreshDuration: credRefreshInterval,
+	}
+
+	rootProvider, err := pemfile.NewProvider(rootOptions)
+	if err != nil {
+		fmt.Printf("Error %v\n", err)
+		os.Exit(1)
+	}
+	return rootProvider
+}
+
+func makeIdentityProvider(useRevokedCert bool, credsDirectory string) certprovider.Provider {
+	certFilePath := ""
+	if useRevokedCert {
+		certFilePath = filepath.Join(credsDirectory, "server_cert_revoked.pem")
+	} else {
+		certFilePath = filepath.Join(credsDirectory, "server_cert.pem")
+	}
+	identityOptions := pemfile.Options{
+		CertFile:        certFilePath,
+		KeyFile:         filepath.Join(credsDirectory, "server_key.pem"),
+		RefreshDuration: credRefreshInterval,
+	}
+	identityProvider, err := pemfile.NewProvider(identityOptions)
+	if err != nil {
+		fmt.Printf("Error %v\n", err)
+		os.Exit(1)
+	}
+	return identityProvider
+}
+
+func makeCrlProvider(crlDirectory string) *advancedtls.FileWatcherCRLProvider {
+	options := advancedtls.FileWatcherOptions{
+		CRLDirectory: crlDirectory,
+	}
+	provider, err := advancedtls.NewFileWatcherCRLProvider(options)
+	if err != nil {
+		fmt.Printf("Error making CRL Provider: %v\nExiting...", err)
+		os.Exit(1)
+	}
+	return provider
+}
+
+func main() {
+	credentialsDirectory := flag.String("credentials_directory", "", "Path to the creds directory of this repo")
+	flag.Parse()
+	if *credentialsDirectory == "" {
+		fmt.Println("Must set credentials_directory argument")
+		os.Exit(1)
+	}
+	tlsServers(*credentialsDirectory)
+	insecureServer()
+	for {
+		time.Sleep(1 * time.Second)
+	}
```

Comment:
If you want to block forever you can do `<-make(chan struct{})`.

But also, if you are calling `createAndRun...` from here directly, just don't do the last one in a `go` call, and it will block by virtue of `Serve` blocking?